File: Java/Medium/Accepted/maximum-subarray-sum-with-one-deletion_2.java
class Solution {
    public int maximumSum(int[] arr) {
        
        int max = arr[0];
        int[] dp = new int[arr.length];
        dp[0] = arr[0];
        for (int i = 1; i < arr.length; i++) {
            dp[i] = Math.max(dp[i - 1] + arr[i], arr[i]);
            max = Math.max(max, dp[i]);
        }
        int[] dp2 = new int[arr.length];
        dp2[arr.length - 1] = arr[arr.length - 1];
        for (int i = arr.length - 2; i >= 0; i--) {
            dp2[i] = Math.max(dp2[i + 1] + arr[i], arr[i]);
        }
        for (int i = 1; i < arr.length - 1; i++) {
            max = Math.max(max, dp[i - 1] + dp2[i + 1]);
        }
        return max;
    }
}

File: Rust/Medium/WrongAnswer/maximum-subarray-sum-with-one-deletion_2.rs
impl Solution {
    pub fn maximum_sum(arr: Vec<i32>) -> i32 {
        
        let mut max = arr[0];
        let mut dp = vec![0; arr.len()];
        dp[0] = arr[0];
        for i in 1..arr.len() {
            dp[i] = arr[i] + dp[i - 1].max(0);
            max = max.max(dp[i]);
        }
        let mut sum = 0;
        for i in (1..arr.len()).rev() {
            sum += arr[i];
            max = max.max(sum + dp[i - 1].max(0));
        }
        max
    }
}